File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/TraitRepository.java
package me.nathanp.magiccreatures.model;

import java.util.List;

import androidx.lifecycle.LiveData;

public class TraitRepository {
    private static final String TAG = "TraitRepository";
    private TraitRemoteDao remoteDao;

    public TraitRepository() {
        remoteDao = new TraitRemoteDao();
    }

    public LiveData<List<Trait>> getAllTraits() {
        return remoteDao.getAllTraits();
    }
}

File: MagicHax/android/app/src/main/java/me/nathanp/magiccreatures/model/TraitRepository.kt
package me.nathanp.magiccreatures.model

import androidx.lifecycle.LiveData
import androidx.lifecycle.Transformations

class TraitRepository {
    private val remoteDao = TraitRemoteDao()

    val traitMap: LiveData<Map<String, Trait>>
        get() = Transformations.map(remoteDao.allTraits) { traits ->
            val traitMap: MutableMap<String, Trait> = sortedMapOf()
            for (trait in traits) {
                traitMap[trait.name] = trait
            }
            traitMap
        }
}